course_info/tests/jasmine/controllers/DetailsControllerSpec.js

```diff
@@ -212,49 +214,337 @@ describe('Unit testing DetailsController', function () {
+                    course_id: '789',
+                    course_groups: [
+                        {
+                            name: 'group one',
+                            id: 1
+                        },
+                        {
+                            name: 'group two',
+                            id: 2
+                        }
+                    ],
+                    departments: [
+                        {
+                            name: 'dept1',
+                            id: 1
+                        }
+                    ]
+                },
+                term: {
+                    display_name: 'Summer 2015',
+                    academic_year: '2015',
+                },
+                primary_xlist_instances: [],
+            };
+
+            members = {
+                count: 100
+            };
+        });
+
         it('should handle both successful responses even if they arrive ' +
-            'separately');
+            'separately', function(){
+            courseInstances.instances[ci.course_instance_id] = ci;
+            dc = $controller('DetailsController', {$scope: scope});
+            spyOn(dc, 'handleCourseInstanceResponse');
+            $httpBackend.expectGET(courseInstanceURL).respond(200, JSON.stringify(ci));
+            $httpBackend.expectGET(peopleURL).respond(200, JSON.stringify(members));
+            $httpBackend.flush(2);
+            expect(dc.courseInstance['title']).toEqual(ci.title);
+            expect( dc.courseInstance['members']).toEqual(members.count);
+        });
+
         it('should still handle the course instance data even if the ' +
-            'member data fetch fails');
+            'member data fetch fails', function(){
+            courseInstances.instances[ci.course_instance_id] = ci;
+            dc = $controller('DetailsController', {$scope: scope});
+            spyOn(dc, 'handleCourseInstanceResponse');
+            $httpBackend.expectGET(courseInstanceURL).respond(200, JSON.stringify(ci));
+            $httpBackend.expectGET(peopleURL).respond(500);
+            $httpBackend.flush(2);
+            expect(dc.courseInstance['title']).toEqual(ci.title);
+        });
     });
 
     describe('submitCourseDetailsForm', function() {
-        it('should call backend with only the editable form fields');
+        var ci;
+        var members;
+        var expectedPatchData = {
+                "description":"this is a test",
+                "instructors_display":"test teacher",
+                "location":"test room",
+                "meeting_time":"test oclock",
+                "notes":"test notes",
+                "short_title":"test short title",
+                "sub_title":"test sub title",
+                "title":"test title"
+            };
+
+        beforeEach(function () {
+            ci = {
+                course_instance_id: $routeParams.courseInstanceId,
+                title: 'Test Course Title',
```

Comment:
Maybe we could define (most of this) in a common function/variable somewhere, instead of declaring it twice in the same suite?